File: CPE-1star/UVA10221-Satellites .cpp
#include <iostream>
#include <cmath>
#include <string>
using namespace std;
int main() {
    double s = 0, distance = 0;
    int r, degree;
    string unit;
    while(cin>>r>>degree>>unit){
        if (unit == "min")
            degree /= 60;
        s = 2 * M_PI * (r + 6440) * (degree / 360);
        distance = 2 * (r + 6440) * sin(degree * M_PI / 180 / 2);
        cout << fixed;
        cout.precision(6);
        cout << round(s * 1000000) / 1000000 << " " << round(distance * 1000000) / 1000000 << std::endl;
        s = 0;
        distance = 0;
    }
    return 0;
}

File: CPE-1star/UVA10221-Satellites .py
# 就是一個公式，可以記
# https://www.youtube.com/watch?v=03w6m8uYCqU&list=PLYhy6KLAgQE6QpZ5dUz-p-5dC1bWCTuIi&index=37
import math

s, distance = 0, 0
while True:
    try:
        r, degree, unit = input().split()
        r = int(r)
        degree = int(degree)
        if unit == "min":
            degree = degree / 60
        s = 2 * math.pi * (r + 6440) * (degree / 360)
        distance = 2 * (r + 6440) * math.sin(degree * math.pi / 180 / 2)
        print("%.6f %.6f" % (round(s, 6), round(distance, 6)))
        s, distance = 0, 0
    except EOFError:
        break
